### ISSUE TITLE
Container interaction (kafka) dockertest set up

### ISSUE THREAD

**Issue by drakos74** [ory/dockertest#195]
**Is your feature request related to a problem? Please describe.**

- When trying to use dockertest with multiple containers (i.e.  kafka and zookeeper) . It is not very straightforward how to run this test. The communication of the containers is a bit more involved than the simple case. Also one of the containers does not expose any ports. And finally debugging the containers is not very straightforward (e.g. container logging)

**Describe the solution you'd like**

- Get the ip of container(s) to pass it over to the next one
- define explicitly the exposed ports
- follow the logs of the containers in parallel to the test execution

**Additional context**

I ve set up a test like this, dockertest can fully support such a use case.
I would like to add this solution to the examples section, so that other users find it easier to use kafka or interacting containers when using dockertest.

**Comment by xakep666** [comment]
@drakos74 since v3.6.0 you can connect containers to docker network so it should help with kafka+zookeeper

**Comment by carlows** [comment]
Do you guys have any examples on how to use the library with kafka + zookeper? 🙂

**Comment by odilonjk** [comment]
@carlows Hey man! Not sure if you still looking for an example, however I'll answer with what I found so far, just in case someone else needs it.

I'm ignoring the error returns to simplify the example. Please keep in mind you'll need to handle it.

1 - Create a network
```
network, _ := pool.Client.CreateNetwork(docker.CreateNetworkOptions{Name: "coolest_network_ever"})
```

2 - Create the Zookeeper container (this example uses the confluent container).
See how the `network` and `PortBindings` is used here in the options:
```
z, _ = pool.RunWithOptions(&dockertest.RunOptions{
   Repository: "confluentinc/cp-zookeeper",
   Tag:        "latest",
   NetworkID:  network.ID,
   Hostname:   "zookeeper",
   PortBindings: map[docker.Port][]docker.PortBinding{
      "2181/tcp": {{HostIP: "zookeeper", HostPort: "2181/tcp"}},
   },
   Env: []string{"ZOOKEEPER_CLIENT_PORT=2181"},
})
```
3 - Create the Kafka broker. Pretty similar to the way you did with Zookeeper.
```
broker, _ = pool.RunWithOptions(&dockertest.RunOptions{
   Repository: "confluentinc/cp-server",
   Tag:        "latest",
   NetworkID:  network.ID,
   Hostname:   "broker",
   PortBindings: map[docker.Port][]docker.PortBinding{
      "29092/tcp": {{HostIP: "broker", HostPort: "29092/tcp"}},
      "9092/tcp":  {{HostIP: "localhost", HostPort: "9092/tcp"}},
   },
   Env: []string{
      "KAFKA_BROKER_ID=1",
      "KAFKA_LISTENER_SECURITY_PROTOCOL_MAP=PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT",
      "KAFKA_ADVERTISED_LISTENERS=PLAINTEXT://broker:29092,PLAINTEXT_HOST://localhost:9092",
      "KAFKA_ZOOKEEPER_CONNECT=zookeeper:2181",
   },
})
```

I'll try to open a PR adding a example like this to the doc. I just need to find time to do it. =P

**Comment by fernandodr19** [comment]
@odilonjk I keep getting ``Unable to resolve address: zookeeper:2181`` on kafka container running your example

**Comment by aeneasr** [comment]
I am closing this issue as it has not received any engagement from the community or maintainers in a long time. That does not imply that the issue has no merit. If you feel strongly about this issue

- open a PR referencing and resolving the issue;
- leave a comment on it and discuss ideas how you could contribute towards resolving it;
- open a new issue with updated details and a plan on resolving the issue.

We are cleaning up issues every now and then, primarily to keep the 4000+ issues in our backlog in check and to [prevent](https://www.jeffgeerling.com/blog/2020/saying-no-burnout-open-source-maintainer) [maintainer](https://www.jeffgeerling.com/blog/2016/why-i-close-prs-oss-project-maintainer-notes) [burnout](https://docs.brew.sh/Maintainers-Avoiding-Burnout). Burnout in open source maintainership is a [widespread](https://opensource.guide/best-practices/#its-okay-to-hit-pause) and serious issue. It can lead to severe personal and health issues as well as [enabling catastrophic](https://haacked.com/archive/2019/05/28/maintainer-burnout/) [attack vectors](https://www.gradiant.org/en/blog/open-source-maintainer-burnout-as-an-attack-surface/).

Thank you to anyone who participated in the issue! 🙏✌️
